.chloggen/docker-resource-failure.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
I would want to do this in a non-breaking way, at least initially. Can we make the change opt-in?

Reply:
I don't see a big issue in being this breaking, since the current behavior seems to me more as a bug - not trying to reconnect to the detectors, if the first attempt failed.

The only thing that has changed from a user perspective with `async_detection: true` is the ordering. We can change the default value for `async_detection` to `false`, which will keep the order of execution of configured detectors. But in this case, one detector failing to start can block all others from starting.

Do you think this is critical for the users?

Reply:
adapted to not breaking